VocaDbWeb/Views/Song/Details.cshtml

```diff
@@ -466,16 +466,28 @@
 					</li>
 				}
 			</ul>
-			
-			<div data-bind="with: selectedLyrics">
-				<p data-bind="visible: source || url">
-					(@Res.LyricsFrom
-					<span data-bind="visible: url"><a class="extLink" data-bind="text: source || url, attr: { href: url, onclick: 'app.functions.trackOutboundLink(event)' }"></a>)</span>					
-					<span data-bind="visible: !url, text: source + ')'"></span>
-				</p>
-
-				<p class="song-lyrics" data-bind="text: value"></p>				
-			</div>
+
+			@foreach (var lyrics in Model.Lyrics.OrderBy(l => l.TranslationType))
+			{
+				<div data-bind="visible: selectedLyricsId() == @lyrics.Id">
+					@if (!string.IsNullOrEmpty(lyrics.Source) || !string.IsNullOrEmpty(lyrics.URL))
+					{
+						<p>
+							(@Res.LyricsFrom
+							@if (!string.IsNullOrEmpty(lyrics.URL))
+							{
+								<span><a class="extLink" href="@lyrics.URL" onclick="app.functions.trackOutboundLink(event)">@(!string.IsNullOrEmpty(lyrics.Source) ? lyrics.Source : lyrics.URL)</a>)</span>
+							}
+							else
+							{
+								<span><text>@lyrics.Source</text>)</span>
+							}
+						</p>
+					}
+
+					<p class="song-lyrics">@lyrics.Value</p>

```

Comment:
But now you're including all the lyrics content on the song page, which will potentially add a significant amount of data. Is it worth it just for Google? Did you compare the song page load time?

Reply:
I've reverted the changes.

Reply:
Did you find out if it actually causing performance issues?

Reply:
Even if the difference is just 10ms, you know that if the song has lyrics in multiple languages, it can be hundreds of kb extra transfer. Better to load that lazily. Google counts both data transfer amount and speed. It's not an important issue anyway, and there is a way to make the lyrics appear in Google while keeping the lazy load, which in my opinion is the best way.